### ISSUE TITLE
Question: i need help

### ISSUE THREAD

**Issue by sebby09559** [ninjamuffin99/Funkin#780]
## What is your question?
how do i make a part of a song have the character change to a different character sprite durining a song 
like this for example https://www.youtube.com/watch?v=gv_rpjpAuhk.
like how do i change the character during a song in the level

**Comment by hdboye** [comment]
What i would do is:
1. make spritesheet with both character sprites
2. when the song is playing, check for curBeat or curStep to reach a certain number (when you want the change to happen)
3. then at the code, instead of it going playAnim('idle'); or smthn make an if else statement that checks if its inbetween those cursteps or curbeats
4. if it is then playAnim('idle-alt'); and if it isnt playAnim('idle')

really terrible explanation but you probably get the gist of it if you code

**Comment by C0RPXE** [comment]
Hello! I really really need help with FNF 🙏🏻  Some of the characters are black boxes and not showing up 😭 What can I do?  Please help me, thank You!

**Comment by devin503** [comment]
> Hello! I really really need help with FNF 🙏🏻 Some of the characters are black boxes and not showing up 😭 What can I do? Please help me, thank You!

Try using the correct version for your system. If you're on Windows 32-bit, download the 32-bit version. Windows 64-bit uses the 64-bit. Also, search for your problem. #706, #370, #356, etc... Plenty of people have mentioned your issue in the past.

**Comment by C0RPXE** [comment]
Yes, I am using 64bit and everything works perfectly, just  not some mods.
And thank you for respond! Much support.

**Comment by ExpiredOats** [comment]
> What i would do is:
> 
> 1. make spritesheet with both character sprites
> 2. when the song is playing, check for curBeat or curStep to reach a certain number (when you want the change to happen)
> 3. then at the code, instead of it going playAnim('idle'); or smthn make an if else statement that checks if its inbetween those cursteps or curbeats
> 4. if it is then playAnim('idle-alt'); and if it isnt playAnim('idle')
> 
> really terrible explanation but you probably get the gist of it if you code

would this be in playstate.hx?

**Comment by hdboye** [comment]
> > What i would do is:
> > 
> > 1. make spritesheet with both character sprites
> > 2. when the song is playing, check for curBeat or curStep to reach a certain number (when you want the change to happen)
> > 3. then at the code, instead of it going playAnim('idle'); or smthn make an if else statement that checks if its inbetween those cursteps or curbeats
> > 4. if it is then playAnim('idle-alt'); and if it isnt playAnim('idle')
> > 
> > really terrible explanation but you probably get the gist of it if you code
> 
> would this be in playstate.hx?

Probably, but you'd need to get the beat and step variables from MusicBeatState.

**Comment by Zicolie** [comment]
> > What i would do is:
> > 
> > 1. make spritesheet with both character sprites
> > 2. when the song is playing, check for curBeat or curStep to reach a certain number (when you want the change to happen)
> > 3. then at the code, instead of it going playAnim('idle'); or smthn make an if else statement that checks if its inbetween those cursteps or curbeats
> > 4. if it is then playAnim('idle-alt'); and if it isnt playAnim('idle')
> > 
> > really terrible explanation but you probably get the gist of it if you code
> 
> would this be in playstate.hx?



> What i would do is:
> 
> 1. make spritesheet with both character sprites
> 2. when the song is playing, check for curBeat or curStep to reach a certain number (when you want the change to happen)
> 3. then at the code, instead of it going playAnim('idle'); or smthn make an if else statement that checks if its inbetween those cursteps or curbeats
> 4. if it is then playAnim('idle-alt'); and if it isnt playAnim('idle')
> 
> really terrible explanation but you probably get the gist of it if you code

Hey so I'm super new to modding and stuff and I just wannt know how to know what your curBeat is, I can see in the charting menu what the curStep is but not curBeat, can you inform me of this? Thanks :)
